src/gpl/src/mbff.cpp

```diff
@@ -0,0 +1,1024 @@
+				}
+
+				cout << "Total = " << (response.objective_value()) << "\n";
+
+
+				float TOT_D = 0;
+				for (int i = 0; i < num_flops; i++) {
+					for (int j = 0; j < num_trays; j++) {
+							for (int k = 0; k < (int)trays[j].slots.size(); k++) if (trays[j].cand[k] == i) {
+									TOT_D += (0.01 * SolutionIntegerValue(response, d_x[i][j][k]));
+									TOT_D += (0.01 * SolutionIntegerValue(response, d_y[i][j][k]));
+							}
+					}
+				}
+				cout << "D: " << TOT_D << "\n";
+
+
+				float TOT_Z = 0;
+				for (int i = 0; i < num_paths; i++) {
+					TOT_Z += (0.01 * SolutionIntegerValue(response, z_x[i]));
+					TOT_Z += (0.01 * SolutionIntegerValue(response, z_y[i]));
+				}
+				cout << "Z: " << TOT_Z << "\n";
+
+
+				float TOT_W = 0;
+				for (int i = 0; i < num_trays; i++) {
+					if (SolutionIntegerValue(response, e[i])) {
+						TOT_W += w[i];
+					}
+				}
+				cout << "W: " << TOT_W << "\n";
+
+				int tot_cnt = 0;
+				set<int> tray_idx;
+				for (int i = 0; i < num_flops; i++) {
+						for (int j = 0; j < num_trays; j++) {
+								for (int k = 0; k < (int)trays[j].slots.size(); k++) if (trays[j].cand[k] == i) {
+										if (SolutionIntegerValue(response, B[i][j][k]) == 1) {
+												tray_idx.insert(j);
+												tot_cnt++;
+										} 
+								}
+						}
+				}
+
+
+				map<int, int> tray_sz;
+				for (auto x: tray_idx) {
+						tray_sz[(int)trays[x].slots.size()]++;
+				}
+
+				for (auto x: tray_sz) {
+						cout << "Tray size = " << x.first << ": " << x.second << "\n";
+				}
+
+				if (tot_cnt != num_flops) {
+					cout << "ERROR -- NOT ALL FLOPS ARE MATCHED\n";
+				}
+
+				cout << "\n";
+
+			} // end ILP 
+
+
+	float MBFF::GetDist(Point a, Point b) {
+		return (abs(a.x - b.x) + abs(a.y - b.y));
+	}
+
+	int MBFF::GetRows(int K) {
+		if (K == 1 || K == 2 || K == 4) return 1;
+		if (K == 8) return 2;
+		else return 4;
+	}
+
+
+	vector<Point> MBFF::GetSlots(Point tray, int rows, int cols) {
+		int bit_idx = 0;
+		for (int i = 1; i < 6; i++) {
+			if (rows * cols == BITCNT[i]) bit_idx = i;
+		}
+
+		float center_x = tray.x, center_y = tray.y;
+
+		vector<Point> slots;
+		for (int i = 0; i < rows * cols; i++) {
+			int new_col = i % cols, new_row = i / cols;
+
+			Point new_slot;
+			new_slot.x = center_x + WIDTH * RATIOS[bit_idx] * ((new_col + 0.5) - (cols / 2.0));
+			new_slot.y = center_y + HEIGHT * ((new_row + 0.5) - (rows / 2.0));
+
+
+			if (new_slot.x >= 0 && new_slot.y >= 0) slots.push_back(new_slot);
+		}
+
+		return slots;
+	}
+
+
+	Flop MBFF::GetNewFlop(vector<Flop> prob_dist, float tot_dist) {
+		float rand_num = (float)(mt() % 101), cum_sum = 0;
+	    Flop new_flop;
+		for (int i = 0; i < (int)prob_dist.size(); i++) {
+			cum_sum += prob_dist[i].prob;
+			new_flop = prob_dist[i];
+			if (cum_sum * 100.0 >= rand_num * tot_dist) break;
+		}
+
+		return new_flop;
+	}
+
+	vector<Tray> MBFF::GetStartTrays(vector<Flop> &flops, int num_trays, float AR) {
+
+		int num_flops = (int)flops.size();
```

Comment:
Prefer to use size_t instead of int when dealing with vector sizes. Also if you need to down convert use `static_cast<int>()` instead of c-style casts like `(int)`